contracts/external/traders/Expiry.sol

```diff
@@ -234,6 +265,59 @@ contract Expiry is
 
     // ============ Private Functions ============
 
+    function callFunctionSetExpiry(
+        address sender,
+        bytes memory data
+    )
+        private
+    {
+        (
+            CallFunctionType callType,
+            SetExpiryArg[] memory expiries
+        ) = abi.decode(data, (CallFunctionType, SetExpiryArg[]));
+
+        assert(callType == CallFunctionType.SetExpiry);
+
+        for (uint256 i = 0; i < expiries.length; i++) {
+            SetExpiryArg memory exp = expiries[i];
+            uint32 timeDelta = exp.timeDelta;
+            if (exp.account.owner != sender) {
+                uint32 minApprovedTimeDelta = g_approvedSender[exp.account.owner][sender];
+                if (minApprovedTimeDelta == 0) {
+                    // don't do anything if sender is not approved
+                    continue;
+                } else {
+                    // bound the time by the minimum approved timeDelta
+                    timeDelta = Math.max(minApprovedTimeDelta, exp.timeDelta).to32();
```

Comment:
Hmm... I wonder if this should do nothing instead of setting it to `minApprovedTimeDelta`